misc fixes

notslang opened the pull request:
PS: that coveralls decrease is because most of the tests now just run the commands via shelljs, so it's done in a separate node process that istanbul doesn't track :(

notslang committed:
fix package.json, using fixpack

notslang committed:
add editorconfig rule for makefile

also add a few other important ones

notslang committed:
upgrade accord

notslang committed:
make coffee-script not a hard dep

since we compile JS before publishing anyway, coffeescript shouldn't
need to be installed for the bin to work.

if there was a better way to not load it when the JS is compiled, then
that would be nice...

notslang committed:
get basic compliance with the editorconfig

notslang committed:
basic syntax corrections

notslang committed:
don't require/load all compilers

(when searching for the right file extension)

notslang committed:
get rid of extraneous resolvePath function

notslang committed:
get rid of --out arg for regular use - closes #1

also remove colors dep

notslang committed:
make tests more robust by testing actual commands

notslang committed:
update docs - closes #2

fix watch example
fix spelling
add note about using pipes

notslang committed:
use argparse & pass data to render with --data

notslang committed:
bump coffee-script & istanbul versions

jescalan commented:
This is a lot of separate changes, as usual would definitely prefer this to be split into a few pull requests rather than one gigantic one covering a large variety of changes.

notslang committed:
make compile into positional arg

jescalan commented:
Hey just checking on the status here, would like to get this merged in still

notslang commented:
Yeah, I'm gonna be working on intermedia for a while :frowning:  ... still going to be finishing it, just don't have time right now.

jescalan commented:
Cool, no worries, just checking in!

notslang committed:
ensure marked is above 0.3.1 due to vulnerability

https://nodesecurity.io/advisories/marked_multiple_content_injection_vulnerabilities

jescalan commented:
Hey just want to give you a reminder that, as mentioned in the first comment here, I still am going to want separate pull requests per change rather than one gigantic full rewrite "misc" pull request. I don't want this to turn into the same sticky situation as we are in with the accord PR, so let's try to keep this one on track from the start.

notslang committed:
improve error message for unmatched extensions

notslang committed:
misc

notslang commented:
one PR per commit?

jescalan commented:
Nope, one PR per _idea/feature_. Sometimes this is just one commit, sometimes more. We're pretty good at breaking up commits into logical chunks and submitting PRs based on that around here. For example, a PR called "miscellaneous" with 30 commits is not really useful. If you are having trouble with the logical chunking concept, grab one of us for help as soon as you start to notice things getting out of hand, and we can discuss it and figure it out.

notslang commented:
what logical chunks do you see here?

jescalan commented:
I'd have to spend a bunch of time reviewing the commits and figuring this out. Are you unable to do so? If you are not, I can add this to my todo list.

notslang commented:
Nope, I grouped all of my changes into commits, where each commit is one logical grouping of changes. As a whole they're not really working towards any particular feature, they're just a set of miscellaneous improvements.

Though I'd like you to review the commits anyway & make sure there aren't any bugs or things I missed.

jescalan commented:
If that was your approach, then yes I'd like one PR per commit. When you submit all the commits in a single PR, I can either take or leave all of them at the same time. When they are in separate PRs, we can merge in ones that we both agree on and get them shipped right away, and for ones that are more conflicted, there is a place for discussing just those conflicts.

notslang commented:
Oh, you can use [git cherrypick](http://git-scm.com/docs/git-cherry-pick) to accept individual commits.

jescalan commented:
I know that. But I'd rather use github pull requests for their strengths here, and they are very good at providing an area to discuss a proposed change to code and merging those changes with a breadcrumb trail back to where they were proposed and the discussion on them.

notslang commented:
I'm fine with using the line comments for discussing individual commits... I don't think I see the problem you're getting at.

jescalan commented:
I'm not fine with that. I would rather have a pull request per feature.

notslang commented:
what's your issue with it? maybe we could find a way around it so I don't need to make 17 separate pull requests?

jescalan commented:
https://github.com/carrot/accord-cli/pull/3#issuecomment-69814119 <-- I like this functionality and prefer to use it. I'm sorry about the 17 pull requests, but the good news is that next time we can start from the beginning submitting a pull request per feature and you don't have to do this separation again.

notslang commented:
You can definitely do that on any comment. For example: https://github.com/slang800/accord-cli/commit/24499e1977fad0a0dec70729d255eaf933c94a1c#commitcomment-7541742

You just right-click on the date & select "copy link address"

jescalan commented:
I would rather use pull requests. I'm sorry. I know it's a pain. But we have to do it.
